lib/class-wp-rest-plugins-controller.php

```diff
@@ -0,0 +1,829 @@
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Updates one plugin.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function update_item( $request ) {
+		require_once ABSPATH . 'wp-admin/includes/plugin.php';
+
+		$data = $this->get_plugin_data( $request['plugin'] );
+
+		if ( is_wp_error( $data ) ) {
+			return $data;
+		}
+
+		$status = $this->get_plugin_status( $request['plugin'] );
+
+		if ( $request['status'] && $status !== $request['status'] ) {
+			$handled = $this->handle_plugin_status( $request['plugin'], $request['status'], $status );
+
+			if ( is_wp_error( $handled ) ) {
+				return $handled;
+			}
+		}
+
+		$this->update_additional_fields_for_object( $data, $request );
+
+		$request['context'] = 'edit';
+
+		return $this->prepare_item_for_response( $data, $request );
+	}
+
+	/**
+	 * Checks if a given request has access to delete a specific plugin.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has access to delete the item, WP_Error object otherwise.
+	 */
+	public function delete_item_permissions_check( $request ) {
+		if ( ! current_user_can( 'activate_plugins' ) ) {
+			return new WP_Error(
+				'rest_cannot_manage_plugins',
+				__( 'Sorry, you are not allowed to manage plugins for this site.', 'gutenberg' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		if ( ! current_user_can( 'delete_plugins' ) ) {
+			return new WP_Error(
+				'rest_cannot_manage_plugins',
+				__( 'Sorry, you are not allowed to delete plugins for this site.', 'gutenberg' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		$can_read = $this->check_read_permission( $request['plugin'] );
+
+		if ( is_wp_error( $can_read ) ) {
+			return $can_read;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Deletes one plugin from the site.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function delete_item( $request ) {
+		require_once ABSPATH . 'wp-admin/includes/file.php';
+		require_once ABSPATH . 'wp-admin/includes/plugin.php';
+
+		$data = $this->get_plugin_data( $request['plugin'] );
+
+		if ( is_wp_error( $data ) ) {
+			return $data;
+		}
+
+		if ( is_plugin_active( $request['plugin'] ) ) {
+			return new WP_Error(
+				'rest_cannot_delete_active_plugin',
+				__( 'Cannot delete an active plugin. Please deactivate it first.', 'gutenberg' ),
+				array( 'status' => 400 )
+			);
+		}
+
+		$filesystem_available = $this->is_filesystem_available();
+		if ( is_wp_error( $filesystem_available ) ) {
+			return $filesystem_available;
+		}
+
+		$deleted = delete_plugins( array( $request['plugin'] ) );
```

Comment:
Should we also have a param here to allow you to also uninstall the plugin at the same time.

Reply:
No, that can break things. See the second half of my comment here: https://github.com/WordPress/gutenberg/pull/22454#issuecomment-631849871